src/jquery.ime.js

```diff
@@ -638,9 +639,9 @@
 			textNode = range.startContainer;
 			textOffset = range.startOffset;
 			textNode.nodeValue =
-				textNode.nodeValue.substr( 0, textOffset - precedingCharCount ) +
+				textNode.nodeValue.slice( 0, Math.max( 0, textOffset - precedingCharCount ) ) +
 				newText +
-				textNode.nodeValue.substr( textOffset );
+				textNode.nodeValue.slice( textOffset );
```

Comment:
Automatic change by eslint, which requires slice() instead of substring() and substr().